src/tests/ilverify/ILTests/ExceptionRegionTests.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+public class ExceptionRegionTests
+{
+    public void ExceptionDerivedType_Valid()
+    {
+        try
+        {
+            int x = 42;
```

Comment:
We should have all the tests for this in IL. I am sorry if I have not said that clearly.

For example, C# compiler can choose to optimize this whole method into no-op and then it won't test what it is supposed to anymore.

Reply:
It was in IL before, I just moved it to C#. Do you mean we should only put `throw null` and `throw InvalidOperationException` cases in C#?

Reply:
I think all test cases for ILVerify should stay in IL.

Reply:
> I think the rule for these sanity checks should be that they never trigger on anything produced by C# compiler.

Sorry I misinterpreted this. :) I thought we still need to add positive test case for this, and to do that we need csc generated assembly.